src/articles/2024-01-21_DDEV-for-js.md

```diff
@@ -0,0 +1,327 @@
+
+As a front-end developer with no interest or experience in DevOps, the key feature for _why_ you should consider DDEV is standardized environments for every member of a team working on the same project.
+
+## Why would I do this instead of a Node/Python server?
+
+If you're successfully developing with some other solution running on a localhost port, you might be fine sticking with that. In my experience, however, there are a couple of significant downsides and painpoints to working from a node process running in the terminal:
+
+1. The process has to continue un-interrupted. If you're an obsessive tab-closer like me, you've probably accidentally quit a running process once or twice. It's not a huge problem to start the process back up, but if you're running the process inside a container you can't accidentally quit it.
+
+2. If you need to switch between several projects using the same tech stack they might all be configured to run on the same port. Again, not an issue as many processes will detect the in-use port and suggest an alternative, but as you move to the browser, all your projects are running on very similar ports, and they might not be the same port every visit. With DDEV, each project gets a URL appended to your local DNS records which points to a router container. Your projects will _never_ conflict, and their URLs will _never_ change.
+
+3. Similar to #2, if you're working on a project with a decoupled front and back-end, your project probably has some CORS configuration setup. With DDEV, you can add the DDEV-generated URL to that CORS config and not have to worry about ports changing over time.
+
+## How do I start?
+
+_Heads up from future me: I'm working on a ddev addon that will handle **a lot** of this work for you. It can be found [here](https://github.com/andy-blum/ddev-pm2)_
+
+Your first step is to install DDEV and its dependencies to your host machine. There's a pretty solid set of instructions [in the docs](https://ddev.readthedocs.io/en/stable/users/install/ddev-installation/).
+
+Once you've got DDEV ready to go, navigate to the root of your project. For the purposes of this article, we'll be working on a hypothetical project using [Keystone CMS](https://keystonejs.com/) for the backend and [Sveltekit](https://kit.svelte.dev/) for the frontend.
+
+If you'd like to have actual running apps for both of these to see everything working, use each app's quickstart commands:
+
+```txt
+$ npm create keystone-app keystone
+```
+
+```txt
+$ npm create svelte@latest svelte
+```
+
+Both of these commands will end with instructions to start the applications, but we'll skip that for now. Our project structure should now look like this:
+
+```txt
+my-project
+|- keystone
+|- svelte
+```
+
+From `my-project`, run `ddev config`.
+
+```txt
+$ ddev config
```

Comment:
I think I'd just do `ddev config --auto` here since they don't care about anything it discovers.

Or consider `ddev config --additional-hostnames=api.my-project,app.my-project`, but that may be skipping too many tutorial steps.

Reply:
When you write this, also worth calling out that you'd commit `.ddev` to your project repo, so this step is only needed for one developer to do once.